### DISCUSSION TITLE
Full coverage of Solution

### DISCUSSION THREAD

**Discussion by AlbertoMonteiro** [coverlet-coverage/coverlet#1498] [Q&A] [answered]
Is there any way to include in coverage even the projects that are not reference but they are in the solution??

The given example: https://github.com/AlbertoMonteiro/CoverageSample

Its generates 100% coverage, but I have 2 class lib projects and one test project, the test project only reference one of class lib, and whe I run `dotnet test --collect:"XPlat Code Coverage"` then I use reportgenerator to get the html report using `dotnet reportgenerator -reports:"**/coverage.cobertura.xml" -targetdir:"coverage" -reporttypes:"HTML;TextSummary"` it shows 100% coverage.

But I wanted that the other project that doesnt have any test to be showed as well, so for that sample would be 50% code coverage instead 100%.

![image](https://github.com/coverlet-coverage/coverlet/assets/836496/8d2ed9d6-29d7-46ee-a79a-5766a84bfb1c)

**Comment by daveMueller** [comment, marked as answer]
Yes you can with the include filters. For instance if you use the `--include-directory` filter with the global tool like this:
```powershell
"D:\Repos\CoverageSample1498\test\CoverageSample.TestProject\bin\Debug\net7.0\CoverageSample.TestProject.dll" --target "dotnet" --targetargs "test D:\Repos\CoverageSample1498\test\CoverageSample.TestProject --no-build" --include-directory "D:\Repos\CoverageSample1498\src\CoverageSample.ProjectB\bin\Debug\net7.0"
```
you get the following result:

![grafik](https://github.com/coverlet-coverage/coverlet/assets/19378678/6085845c-fb29-432e-a6c1-21ce6a7dceba)

The same also works with the collector. For instance like:
```powershell
dotnet test --collect:"XPlat Code Coverage" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.IncludeDirectory="D:\Repos\CoverageSample1498\src\CoverageSample.ProjectB\bin\Debug\net7.0"
```
you get a result like

![grafik](https://github.com/coverlet-coverage/coverlet/assets/19378678/df6b7182-fd8c-40e2-80d6-e1b2968cd3e1)

You can find some more infos about the filters here:
[coverlet.console](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/GlobalTool.md#filters)
[coverlet.msbuild](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/MSBuildIntegration.md#filters)
[coverlet.collector](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md#advanced-options-supported-via-runsettings)

**Reply by AlbertoMonteiro** [reply]
Dave tyvm for the answer. I used the collector alternative, worked pretty nice!
